src/toM3u8.js

```diff
@@ -0,0 +1,83 @@
+const formatVideoPlaylist = ({ attributes, segments}) => {
+  return {
+    attributes: {
+      NAME: attributes.id,
+      RESOLUTION: {
+        width: parseInt(attributes.width, 10),
+        height: parseInt(attributes.height, 10)
+      },
+      CODECS: attributes.codecs,
+      BANDWIDTH: parseInt(attributes.bandwidth, 10),
+      ['PROGRAM-ID']: 1
+    },
+    uri: '',
+    timeline: attributes.periodIndex,
+    resolvedUri: '',
+    segments
+  };
+};
+
+const formatAudioPlaylist = ({ attributes, segments}) => {
+  return {
+    attributes: {
+      NAME: attributes.id,
+      BANDWIDTH: parseInt(attributes.bandwidth, 10),
+      CODECS: attributes.codecs,
+      ['PROGRAM-ID']: 1
+    },
+    uri: '',
+    timeline: attributes.periodIndex,
+    resolvedUri: '',
+    segments
+  };
+};
+
+// TODO
+const formatVttPlaylist = playlist => playlist;
+
+export const toM3u8 = dashPlaylists => {
+  if (!dashPlaylists.length) {
+    return {};
+  }
+
+  // grab all master attributes
+  const { sourceDuration: duration } = dashPlaylists[0].attributes;
+
+  const videoOnly = ({attributes}) =>
+    attributes.mimeType === 'video/mp4' || attributes.contentType === 'video';
+  const audioOnly = ({attributes}) =>
+    attributes.mimeType === 'audio/mp4' || attributes.contentType === 'audio';
+  const vttOnly = ({attributes}) =>
+    attributes.mimeType === 'text/vtt' || attributes.contentType === 'text';
+
+  const videoPlaylists = dashPlaylists.filter(videoOnly).map(formatVideoPlaylist);
+  const audioPlaylists = dashPlaylists.filter(audioOnly).map(formatAudioPlaylist);
+
+  // TODO
+  const vttPlaylists = dashPlaylists.filter(vttOnly).map(formatVttPlaylist);
+
+  return {
+    allowCache: true,
+    discontinuityStarts: [],
+    segments: [],
+    mediaGroups: {
+      AUDIO: {
+        main: {
```

Comment:
This is a bit different from the out put of the m3u8 parser. Have we figured out how we plan to handle matching the video and audio playlists?
In HLS, this would look like
```
AUDIO: {
  GROUP-ID: {
    NAME: playlist
  }
}
```
example
```
AUDIO: {
  LOW: {
    English: playlist,
    Spanish: playlist
  },
  HIGH: {
    English: playlist,
    Spanish: playlist
  }
}
```

Reply:
Yep that's just a stub, the highest audio rendition for each track will live in a hard coded HIGH group which will link to each of the video playlists. In the future we can see if we could pair low quality audio with video.